src/throttle/amoc_throttle.erl

```diff
@@ -3,57 +3,63 @@
+%% Throttle unit of measurement
+
+-type gradual_rate_config() :: #{from_rate := non_neg_integer(),
+                                 to_rate := non_neg_integer(),
+                                 interval => interval(),
+                                 step_interval => pos_integer(),
+                                 step_size => pos_integer(),
+                                 step_count => pos_integer(),
+                                 duration => pos_integer()} |
+                               #{from_interarrival := pos_integer(),
+                                 to_interarrival := pos_integer(),
+                                 step_interval => pos_integer(),
+                                 step_size => pos_integer(),
+                                 step_count => pos_integer(),
+                                 duration => pos_integer()}.
+%% Configuration for a gradual throttle rate change
+%%
+%% `From' and `To' rates are required. `interval' defaults to 1s and `step_size' to 1 (or -1 if applies),
+%% that is, the throttle will be changed in increments of 1.
+%%
+%% All other values can be calculated from the provided.
 
-%% @see start/4
--spec start(name(), rate(), non_neg_integer()) -> ok | {error, any()}.
-start(Name, Rate, Interval) ->
-    start(Name, Rate, Interval, ?DEFAULT_NO_PROCESSES).
+-export_type([t/0, name/0, rate/0, interval/0, gradual_rate_config/0]).
 
-%% @doc Starts the throttle mechanism for a given `Name' with a given `Rate' per `Interval'.
+%% @doc Starts the throttle mechanism for a given `Name' with a given config.
 %%
-%% The optional arguments are an `Interval' (default is one minute) and a ` NoOfProcesses' (default is 10).
 %% `Name' is needed to identify the rate as a single test can have different rates for different tasks.
-%% `Interval' is given in milliseconds and can be changed to a different value for convenience or higher granularity.
-%% It also accepts a special value of `0' which limits the number of parallel executions associated with `Name' to `Rate'.
--spec start(name(), rate(), interval(), pos_integer()) -> ok | {error, any()}.
-start(Name, Rate, Interval, NoOfProcesses)
-  when is_atom(Name), ?POS_INT(Rate), ?NONNEG_INT(Interval), ?POS_INT(NoOfProcesses) ->
-    amoc_throttle_controller:ensure_throttle_processes_started(Name, Rate, Interval, NoOfProcesses);
-start(_Name, _Rate, _Interval, _NoOfProcesses) ->
-    {error, invalid_throttle}.
+%% `Interval' is given in milliseconds, the default is one minute,
+%% and can be changed to a different value for convenience or higher granularity.
+-spec start(name(), t() | rate()) -> {ok, started | already_started} | {error, any()}.
+start(Name, #{} = Config) ->
+    amoc_throttle_controller:ensure_throttle_processes_started(Name, Config);
+start(Name, Rate) ->
+    amoc_throttle_controller:ensure_throttle_processes_started(Name, #{rate => Rate}).
 
 %% @doc Pauses executions for the given `Name' as if `Rate' was set to `0'.
```

Comment:
please add a comment that `resume/1` interface is the one to be used to undo pausing.